### PR TITLE
QMAPS-517 Filter local storage on POIs for the favorites panel

### PR DESCRIPTION
**Pull request by patfrat**
Add isCompliantKey method on Poi in order to take in account only POIs in Favorites panel

### CODE REVIEW

File: src/panel/favorites_panel.js [hunk 1]
```diff
@@ -97,13 +97,21 @@ Favorite.prototype.connectStore = async function () {
 
 Favorite.prototype.getAll = async function () {
   let storedData = {}
+  let poisData = []
   try {
     storedData = await this.store.getAll()
+    // check if all data are Pois
+    Object.keys(storedData).map((mapPoint) => {
```
[the remark this answers is not in the document]
Comment (patfrat) [reply src/panel/favorites_panel.js]: ok, i will create a method in local_store > getAllPOIs

File: src/mapbox/poi.js [hunk 2]
```diff
@@ -30,6 +30,11 @@ Poi.prototype.getKey = function () {
   return `qmaps_v${version}_favorite_place_${this.id}`
 }
 
+Poi.isPoiCompliantKey = function (k) {
+  const keyPattern = /qmaps_v[0-9]_favorite_place_*/
```
Comment (Julien-laville) [inline src/mapbox/poi.js]: You should use the version number from config, it's already imported in this file (import {version} from '../../config/constants.yml')

File: src/libs/local_store.js [hunk 3]
```diff
@@ -1,9 +1,11 @@
...
 
-LocalStore.prototype.getAll = async function () {
+LocalStore.prototype.getAllPois = async function () {
   return new Promise((resolve) => {
     try {
-      const items = Object.keys(localStorage).map((k) => {
+      const items = Object.keys(localStorage).filter((k) => {
+        return Poi.isPoiCompliantKey(k)
+      }).map((k) => {
         return JSON.parse(localStorage.getItem(k))
```
Comment (Julien-laville) [inline src/libs/local_store.js]: try catch is too large, it may hide others error than json parse error.

File: src/libs/local_store.js [hunk 4]
```diff
@@ -1,9 +1,11 @@
 function LocalStore() {}
 
-LocalStore.prototype.getAll = async function () {
+LocalStore.prototype.getAllPois = async function () {
```
Comment (Julien-laville) [inline src/libs/local_store.js]: i'm not sure but when you declare a function async it already return promise. i think you can simplify by just Promise.resolve(JSON.parse...)

File: src/adapters/store.js [hunk 5]
```diff
@@ -18,9 +18,9 @@ function Store() {
   })
 }
 
-Store.prototype.getAll = async function() {
+Store.prototype.getAllPois = async function() {
   return new Promise((resolve, reject) => {
-    abstractStore.getAll().then((masqData) => {
+    abstractStore.getAllPois().then((masqData) => {
```
Comment (Julien-laville) [inline src/adapters/store.js]: i have moved from then to async await syntax. can you update this part with the new syntax

File: src/mapbox/poi.js (hunk 2 above)
Comment (patfrat) [reply src/mapbox/poi.js]: ok for that with RegExp

File: src/libs/local_store.js (hunk 4 above)
Comment (patfrat) [reply src/libs/local_store.js]: The async declaration is not from my changes but was already here ?!

File: src/adapters/store.js (hunk 5 above)
Comment (patfrat) [reply src/adapters/store.js]: I prefer to refactor the code in another context rather than do it here

File: src/libs/local_store.js (hunk 4 above)
Comment (Julien-laville) [reply src/libs/local_store.js]: i say that i think that if you declare method with async, you dont have to return a new Promise

File: src/adapters/store.js [hunk 6]
```diff
@@ -18,31 +18,29 @@ function Store() {
...
-      }
-    })
-  })
+  try {
+    let data = await abstractStore.getAllPois()
+    return data != null
+  } catch (error) {
+    if(error.message === 'UNREGISTERED') {
+      return false
+    }
```
Comment (Julien-laville) [inline src/adapters/store.js]: this method let an possible unsolvable promise (in the case of error.message != 'unregistred') maybe we can delete the message test and return false every time we enter in the catch.

File: src/libs/local_store.js [hunk 7]
```diff
@@ -14,18 +17,18 @@ LocalStore.prototype.getAll = async function () {
   })
 }
 
-LocalStore.prototype.register = async function() {
+LocalStore.prototype.register = function() {
   console.log('local storage doesn\'t support register method')
   return new Promise((resolve)=>{resolve()})
```
Comment (Julien-laville) [inline src/libs/local_store.js]: can be smplified ( new Promise((resolve)=>{resolve()})  = Promise.resolve() )

File: src/libs/local_store.js [hunk 8]
```diff
@@ -14,18 +17,18 @@ LocalStore.prototype.getAll = async function () {
...
 }
 
-LocalStore.prototype.onConnect = async function () {
+LocalStore.prototype.onConnect = function () {
   return new Promise((resolve) => {
     resolve()
```
Comment (Julien-laville) [inline src/libs/local_store.js]: can be simplified by return Promise.resolve()

File: tests/integration/tests/favorite.js [hunk 9]
```diff
@@ -38,7 +39,7 @@ test('restore favorite from localStorage', async () => {
   expect.assertions(1)
   await page.goto(APP_URL)
   const testTitle = 'demo_fav'
-  store(page, 'demo_fav', {id : 0, name : testTitle})
+  store(page, 'qmaps_v' + version + '_favorite_place_demo_fav', {id : 0, name : testTitle})
```
[the remark this answers is not in the document]
Comment (patfrat) [reply tests/integration/tests/favorite.js]: ok

File: src/adapters/store.js [hunk 10]
```diff
@@ -18,31 +18,29 @@ function Store() {
...
-      resolve(masqData)
-    }).catch(function (error) {
-      fire('error_h' , 'store ' + error)
-      reject(error)
-    })
-  })
+Store.prototype.getAllPois = async function() {
+  try {
+    let masqData = await abstractStore.getAllPois()
+    return masqData
```
[the remark this answers is not in the document]
Comment (patfrat) [reply src/adapters/store.js]: right !

**Comment by Julien-laville** [comment]
Thank's @patfrat !
🌈 🦉

**Pull request merged**